static/canvas.js

```diff
@@ -126,8 +128,27 @@ class Canvas {
         /* Event for moving the mouse. Primarily used for enlarging a box and moving it. */
         canvas.addEventListener('mousemove', function (e) {
             var mouse = myState.getMouse(e);
-            if (!myState.enlargeDirection && !myState.dragging) {
-                return;
+            var mx = mouse.x;
+            var my = mouse.y;
+            var boxes = myState.boxes;
+            var l = boxes.length;
+            var border = false;
+            var box = null;
+
+            /* Checks if mouse is hovering over a border. */
+            for (var i = l - 1; i >= 0; i--) {
+                box = boxes[i]; 
+                border = boxes[i].withinBorder(mx, my);
+                if (border) {
+                    break;
+                }
+            }
+
+
+            if (!myState.enlargeDirection && !myState.dragging && !border) {
+                document.body.style.cursor = 'default';
+            } else if (!myState.enlargeDirection && !myState.dragging && border) {
+                myState.doubleArrow(box, mx, my);
```

Comment:
Sorry, I'll be clearer. You're right about that, but a combination of functions allow a fix for this. Firstly, this is what I intended when I meant my suggestion.
`
    doubleArrow(border) {
        if (border === Box.border.TOPLEFT || border === Box.border.BOTTOMRIGHT) {
            document.body.style.cursor = 'nwse-resize';
        } else if (border === Box.border.TOPRIGHT || border === Box.border.BOTTOMLEFT) {
            document.body.style.cursor = 'nesw-resize';
        } else if (border === Box.border.TOP || border === Box.border.BOTTOM){
            document.body.style.cursor = 'ns-resize';
        } else if (border === Box.border.LEFT || border === Box.border.RIGHT) {
            document.body.style.cursor = 'ew-resize';
        }
    }
`

You are right that this would cause the cursor to change even though though the mouse doesn't `contains` the shape, but by using the Box.contain function before calling the `doubleArrow` function makes it work and much cleaner. You can even simplify the withinTop and other functions by putting a `contains` check there. You can even call contains in the doubleArrow function. 

If you don't care for either of those suggestions, that's fine, but it'd be nice if you could clean up the conditions in doubelArrow. Thanks!